### ISSUE TITLE
IRI.getFragment() doesn't return always return URI fragments when it could do so

### ISSUE THREAD

**Issue by matthewhorridge** [owlcs/owlapi#199]
Given, ```http://stuff.com/ontology#MyClass(X)``` the value returned by ```IRI.getFragment()``` is ```null``` .  
This has an impact on ```SimpleShortFormProvider``` (well ```SimpleIRIShortFormProvider```) that will return "ontology" as the short form rather than the expected "MyClass(X)".  Brackets are actually allowed in IRI fragments.  

Of course, the problem is, ```IRI.getFragment``` has never done "what is says on the tin".  The method should probably be replaced with some other, better documented methods, and it should be deprecated.

Grammar for fragments
http://tools.ietf.org/html/rfc3986

**Comment by tgbugs** [comment]
Hi all,
   I've been trying to figure out whether there was an existing issue for the problem I'm encountering and I this and #204 seem to be most related to it.

When I load some ttl files with a specified prefix and then save them again the PREFIX:fragement version from the original file is not serialized but instead the full IRI is inserted in its place. An example of a file that does this is [here](https://github.com/tgbugs/methodsOntology/blob/master/ttl/hbp_abam_ontology.ttl), the issue occurs with owlapi v3 in the official protege release and in the v4 version that @ignazio1977 has [here](https://github.com/ignazio1977/protegetests/blob/master/protege-distribution-5.0.0-beta-18-SNAPSHOT-platform-independent.zip). An example of the switch:
before: ```ABA:1```
after: ```<http://api.brain-map.org/api/v2/data/Structure/1>```

Is this issue related? Thanks!

**Comment by ignazio1977** [comment]
Not strictly related - the IRI splitting is only secondary here, since the problem of QNames only exists in RDF/XML. I think in this case the issue is that the namespace abbreviations in your input are not kept. This was the subject of another change (can't recall it now) and might have been fixed in 4.0.2.
Can you raise yours as a separate issue? I'll see if it's reproducible in the latest OWL API and if it's possible for Turtle formats to use a looser form of IRI splitting.

**Comment by tgbugs** [comment]
Will do thanks.
